### PR TITLE
Fix platform checkout Store API nonce check

### PR DESCRIPTION
**Pull request by rafaelzaleski**
#### Changes proposed in this Pull Request

This PR removes the code that disables the store API nonce check for the platform checkout.

#### Testing instructions

- Checkout on `fix/store-api-authentication` in the platform checkout repository
- Make sure you can perform a transaction using the platform checkout

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) : _Add link here / 'QA Testing Not Applicable'_

### CODE REVIEW

File: tests/unit/test-class-wc-payments.php [hunk 1]
```diff
@@ -107,7 +102,7 @@ public function test_rest_endpoints_validate_nonce_if_platform_checkout_feature_
 		$this->assertEquals( 'woocommerce_rest_missing_nonce', $response->get_data()['code'] );
 	}
 
-	public function test_rest_endpoints_do_not_validate_nonce_if_platform_checkout_feature_flag_is_enabled() {
+	public function test_rest_endpoints_do_validate_nonce_if_platform_checkout_feature_flag_is_enabled() {
 		// Enable dev mode so nonce check is disabled.
 		add_filter(
 			'wcpay_dev_mode',
```
Comment (asumaran) [inline tests/unit/test-class-wc-payments.php:108]: Should we remove this? It was added by @reykjalin in #4115 when the "disable nonce check" was introduced.

**Review by asumaran — COMMENTED** [review]
Tests well. I leaving just a comment for you @rafaelzaleski

File: tests/unit/test-class-wc-payments.php (hunk 1 above)
Comment (rafaelzaleski) [reply tests/unit/test-class-wc-payments.php:108]: This was a quick fix to keep the transactions working on our dev environments without adding a potential security risk to live environments. This is no longer needed now that the nonce verification works (dev or live mode).
This PR you are reviewing is a continuation of @reykjalin work.
p1649426189773499-slack-C022WMN88KG

Comment (asumaran) [reply tests/unit/test-class-wc-payments.php:108]: > This is no longer needed now that the nonce verification works

So, should we remove the code: 

```php
add_filter(
	'wcpay_dev_mode',
	function () {
		return true;
	}
);
```

Comment (rafaelzaleski) [reply tests/unit/test-class-wc-payments.php:108]: yes, I'll remove it. Thank you for spotting that.

**Comment by rafaelzaleski** [comment]
@asumaran, I don't know why some checks are failing here. The tests are running well locally, and all I did since the last time was remove three unit tests. 🤔

**Comment by asumaran** [comment]
@rafaelzaleski yeah, it seems to be failing for other PRs too.

**Review by asumaran — APPROVED** [review]
I'm approving this now. Since the tests are failing for unrelated causes I think it should be good to merge this one.

**Pull request merged**